### ISSUE TITLE
GUI startup stays at white blank screen.

### ISSUE THREAD

**Issue by ddkn** [spirit-code/spirit#569]
Hello,

I am compiling spirit with the default options on Debian/Buster amd64 (GCC 8.3, Qt 5.11.3, cmake 3.13.4, python 3.7.3). As for as I know the compilation builds with no known errors. However running the GUI just pops up a white screen
![white screen](https://user-images.githubusercontent.com/13819239/82464393-6b0be380-9a94-11ea-9724-6539e904448a.png)
, and just sits there. I get the following error which is not present in the `Log_{date}.txt` files in `~/tmp/spirit`:

`composeAndFlush: makeCurrent() failed`

That just repeats continuously until I close the window.

I tried building with branches: `master`, `v2.0.1`, `develop`, and `improved-cmake`.

I am just trying to run it in the repository folder with no luck. Install path `/opt/spirit` is also generating the same issue.

I have attached the ldd of `~/tmp/spirit/spirit` and the Log output from running the program.
[debian-ldd.txt](https://github.com/spirit-code/spirit/files/4657723/debian-ldd.txt)
[Log_2020-05-20_12-09-08.txt](https://github.com/spirit-code/spirit/files/4657724/Log_2020-05-20_12-09-08.txt)

Any help on moving forward would be greatly appreciated!

Cheers,

PS. I am running this through ssh -X for the screen shot, however locally on the machine the same issue happens.

PPS. What compile options for the pip version 2.0.1 were set? Is it the same as the base install? Or does it have Threads/OpenMP?

**Comment by ddkn** [comment]
By the way, our lab has a computer that doesn't allow us to install Ubuntu (it locks up on install), so I am using Debian. 

However, on my OSX machine, I have a virtual environment with Ubuntu 20.04. It compiled `improved-cmake` just fine and graphics work as expected with `-DSPIRIT_USE_OPENMP=ON`.

So it seems there is some draw call issues with Qt-5.11.3 on Debian/Buster.

**Comment by GPMueller** [comment]
Yes, the error indicates that it is not able to get the current OpenGL context. I don't know why that would be the case, unless for some reason your driver doesn't support OpenGL 3.3 or newer.

If Qt manages to get an OpenGL context, a line like the following should appear in the log:
```
2020-05-21 11:35:58  [ INFO  ] [UI  ] [--] [--]  QSurfaceFormat version: 3.3
```

**Comment by ddkn** [comment]
Hm, It doesn't get stored in the log file, but shows up in the text. I tried upgrading drivers which broke the server (spent the rest of the morning getting back to a working state). I am using nvidia 418.133 with the both cuda repositories 10.0 and 10.1 installed. Recent updates to debian backports broke a few things it looks like.

![Screen Shot 2020-05-21 at 2 57 40 PM](https://user-images.githubusercontent.com/13819239/82590280-cfea3b00-9b73-11ea-9a04-9bcb4986b7e0.png)

Since we have a non-standard install of drivers for cuda-10.0 to work, I am going to try and petition our lab to upgrade the hardware so we be more in line with Ubuntu. That should make maintenance life night and day easier.

I think these issues are not worth pursuing on this patchy installation. Seeing as spirit worked fine on Ubuntu 20.04 I will try and follow up with a new server.

Thanks for the help.

PS. On the Ubuntu VM I tested, `make intstall` worked as expected.
